lib/smalloc.js

```diff
@@ -3,4 +3,4 @@
 const util = require('internal/util');
 
 module.exports = require('internal/smalloc');
-util.printDeprecationMessage('smalloc is deprecated.');
+util.deprecate('smalloc is deprecated. Use typed arrays.');
```

Comment:
I don't think it will work here. The idea is to print a deprecation message when this line is executed.
`printDeprecationMessage` should stay but we have a problem with the `--trace-deprecation` flag. The actual caller location is where `require('smalloc')` is done.